### PR TITLE
feat: show measurement database and retention policy wildcards

### LINKED ISSUE #22390: Forward port to master branch: show measurements wildcards
Forward port of https://github.com/influxdata/influxdb/issues/3318

### PR DESCRIPTION
**Pull request by lesam**
Closes #22390

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] Documentation updated or issue created: https://github.com/influxdata/docs-v2/issues/3201

### CODE REVIEW

**Comment by lesam** [comment]
Requires https://github.com/influxdata/influxql/pull/47

File: influxql/v1tests/server_test.go [hunk 1]
```diff
@@ -4718,12 +4720,75 @@ func TestServer_Query_ShowMeasurements(t *testing.T) {
...
 		&Write{data: strings.Join(writes, "\n")},
+		&Write{bucketID: bucket2.ID, data: strings.Join(rp1Writes, "\n")},
+		&Write{bucketID: bucket3.ID, data: strings.Join(db1Writes, "\n")},
 	}
 
 	test.addQueries([]*Query{
+		{
+			name:    `show measurements`,
+			command: "SHOW MEASUREMENTS",
+			// *unlike* 1.x, InfluxDB 2 shows measurements from the default retention policy when the rp is not specified, not all retention policies
```
Comment (lesam) [inline influxql/v1tests/server_test.go:4775]: I think this is a bug - https://docs.influxdata.com/influxdb/v2.0/query-data/influxql/ says influxql should be exactly the same as 1.x . I think we should fix this to return measurements from all retention policies mapped for a given db, similar to 1.x .

File: v1/coordinator/statement_executor.go [hunk 2]
```diff
@@ -385,53 +386,125 @@ func (e *StatementExecutor) executeDropMeasurementStatement(ctx context.Context,
...
+			return mappings[i].Database < mappings[j].Database
+		}
+		return mappings[i].RetentionPolicy < mappings[j].RetentionPolicy
+	})
+
+	for _, mapping := range mappings {
+		names, err := e.TSDBStore.MeasurementNames(ctx, ectx.Authorizer, mapping.BucketID.String(), q.Condition)
+		if err != nil || len(names) == 0 {
+			return ectx.Send(ctx, &query.Result{
+				Err: err,
```
Comment (davidby-influx) [inline v1/coordinator/statement_executor.go:450]: Should we separate these cases, `err != nil` and `len(names) == 0` and make up an error for the `len() == 0`?  Or are you counting on the empty query result with a nil error as being interpreted as no error but no data? 

If the latter, would it be cleaner to fall through and have a single exit point for no results (whether by initial return, or by limit and offset?

**Review by davidby-influx — APPROVED** [review]
Non-blocking comment

File: v1/coordinator/statement_executor.go (hunk 2 above)
Comment (lesam) [reply v1/coordinator/statement_executor.go:450]: Drat this is a bug - will fix

Comment (lesam) [reply v1/coordinator/statement_executor.go:450]: @davidby-influx great catch, thanks!

**Review by davidby-influx — APPROVED** [review]
Fixed!

**Comment by est** [comment]
```
 $ influx -database mydb -precision rfc3339
Connected to http://localhost:8086 version 1.8.10
InfluxDB shell version: 1.8.10
> SHOW MEASUREMENTS ON *.*
ERR: error parsing query: found *, expected identifier at line 1, char 22
```

Is this feature not on 1.8.10 yet?

**Comment by davidby-influx** [comment]
@est - new features are going into the 2.X influxdb. The 1.8.X line is only receiving security updates at this point.

**Comment by est** [comment]
> new features are going into the 2.X influxdb

https://docs.influxdata.com/influxdb/v1.8/query_language/explore-schema/#show-measurements

says 

> The following queries with wildcards characters (*) are also allowed:
> SHOW MEASUREMENTS ON *.* to show all databases and all RPs
> SHOW MEASUREMENTS ON <db>.* to show a specific database and all its RPs

I assume it's available on 1.8, but no?

**Comment by davidby-influx** [comment]
@est - looks like a Documentation bug.  Here's my reference from the VP of Product:
https://github.com/influxdata/influxdb/issues/3318#issuecomment-859302318

Let me dig a little deeper to make sure of the status.

**Comment by timhallinflux** [comment]
@est -- what is preventing you from moving to 2.x?

**Comment by est** [comment]
@timhallinflux  I don't have root access on the influxd server 😭 (need great many loops to upgrade)

**Pull request merged**